### PR TITLE
[3] [contrib/metrics] setup typing in contrib part of the library

### LINKED ISSUE #977: Setup Typing in contrib part of the library
## 🚀 Feature

Following #651, idea is to improve contrib code by adding Typing.

### PR DESCRIPTION
**Pull request by RaviTezu**
Fixes #977
Description: Setup Typing in contrib/metrics part of the library

### CODE REVIEW

File: ignite/contrib/metrics/average_precision.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
+from typing import Any, Callable
+
 from ignite.metrics import EpochMetric
 
 
-def average_precision_compute_fn(y_preds, y_targets):
+def average_precision_compute_fn(y_preds: Any, y_targets: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/average_precision.py:6]:
```suggestion
def average_precision_compute_fn(y_preds: torch.Tensor, y_targets: torch.Tensor):
```

File: ignite/contrib/metrics/precision_recall_curve.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
+from typing import Any, Callable
+
 from ignite.metrics import EpochMetric
 
 
-def precision_recall_curve_compute_fn(y_preds, y_targets):
+def precision_recall_curve_compute_fn(y_preds: Any, y_targets: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/precision_recall_curve.py:6]:
```suggestion
def precision_recall_curve_compute_fn(y_preds: torch.Tensor, y_targets: torch.Tensor):
```

File: ignite/contrib/metrics/regression/_base.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ class _BaseRegression(Metric):
     # method `_update`.
 
     @reinit__is_reduced
-    def update(self, output):
+    def update(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/_base.py:39]:
```suggestion
    def update(self, output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/_base.py [hunk 4]
```diff
@@ -21,7 +21,7 @@ def _check_output_shapes(output):
         raise ValueError("Input y should have shape (N,) or (N, 1), but given {}".format(y.shape))
 
 
-def _check_output_types(output):
+def _check_output_types(output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/_base.py:24]:
```suggestion
def _check_output_types(output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/_base.py [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
+from typing import Any, Callable
 
 import torch
 
 from ignite.metrics import EpochMetric, Metric
 from ignite.metrics.metric import reinit__is_reduced
 
 
-def _check_output_shapes(output):
+def _check_output_shapes(output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/_base.py:10]:
```suggestion
def _check_output_shapes(output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/_base.py [hunk 6]
```diff
@@ -50,7 +50,7 @@ def update(self, output):
         self._update((y_pred, y))
 
     @abstractmethod
-    def _update(self, output):
+    def _update(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/_base.py:53]:
```suggestion
    def _update(self, output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/_base.py [hunk 7]
```diff
@@ -59,14 +59,16 @@ class _BaseRegressionEpoch(EpochMetric):
...
-    def __init__(self, compute_fn, output_transform=lambda x: x, check_compute_fn: bool = True):
+    def __init__(
+        self, compute_fn: Callable, output_transform: Callable = lambda x: x, check_compute_fn: bool = True,
+    ):
         super(_BaseRegressionEpoch, self).__init__(
             compute_fn=compute_fn, output_transform=output_transform, check_compute_fn=check_compute_fn
         )
 
-    def _check_type(self, output):
+    def _check_type(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/_base.py:69]:
```suggestion
    def _check_type(self, output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/_base.py [hunk 8]
```diff
@@ -59,14 +59,16 @@ class _BaseRegressionEpoch(EpochMetric):
...
         _check_output_types(output)
         super(_BaseRegressionEpoch, self)._check_type(output)
 
-    def _check_shape(self, output):
+    def _check_shape(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/_base.py:73]:
```suggestion
    def _check_shape(self, output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/canberra_metric.py [hunk 9]
```diff
@@ -34,7 +34,7 @@ def __init__(
     def reset(self):
         self._sum_of_errors = torch.tensor(0.0, device=self._device)
 
-    def _update(self, output):
+    def _update(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/canberra_metric.py:37]:
```suggestion
    def _update(self, output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/fractional_absolute_error.py [hunk 10]
```diff
@@ -24,7 +26,7 @@ def reset(self):
         self._sum_of_errors = 0.0
         self._num_examples = 0
 
-    def _update(self, output):
+    def _update(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/fractional_absolute_error.py:29]:
```suggestion
    def _update(self, output: Tuple[torch.Tensor, torch.Tensor]):
```

File: ignite/contrib/metrics/regression/fractional_bias.py [hunk 11]
```diff
@@ -25,7 +27,7 @@ def reset(self):
         self._sum_of_errors = 0.0
         self._num_examples = 0
 
-    def _update(self, output):
+    def _update(self, output: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/fractional_bias.py:30]:
```suggestion
    def _update(self, output: Tuple[torch.Tensor, torch.Tensor]):
```
and other like that

File: ignite/contrib/metrics/regression/median_absolute_error.py [hunk 12]
```diff
@@ -1,9 +1,11 @@
+from typing import Any, Callable
+
 import torch
 
 from ignite.contrib.metrics.regression._base import _BaseRegressionEpoch
 
 
-def median_absolute_error_compute_fn(y_pred, y):
+def median_absolute_error_compute_fn(y_pred: Any, y: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/median_absolute_error.py:8]: y_pred and y are torch.Tensors

File: ignite/contrib/metrics/regression/median_absolute_percentage_error.py [hunk 13]
```diff
@@ -1,9 +1,11 @@
+from typing import Any, Callable
+
 import torch
 
 from ignite.contrib.metrics.regression._base import _BaseRegressionEpoch
 
 
-def median_absolute_percentage_error_compute_fn(y_pred, y):
+def median_absolute_percentage_error_compute_fn(y_pred: Any, y: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/median_absolute_percentage_error.py:8]: Same here

File: ignite/contrib/metrics/regression/median_relative_absolute_error.py [hunk 14]
```diff
@@ -1,9 +1,11 @@
+from typing import Any, Callable
+
 import torch
 
 from ignite.contrib.metrics.regression._base import _BaseRegressionEpoch
 
 
-def median_relative_absolute_error_compute_fn(y_pred, y):
+def median_relative_absolute_error_compute_fn(y_pred: Any, y: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/regression/median_relative_absolute_error.py:8]: Same here

File: ignite/contrib/metrics/roc_auc.py [hunk 15]
```diff
@@ -1,7 +1,9 @@
+from typing import Any, Callable
+
 from ignite.metrics import EpochMetric
 
 
-def roc_auc_compute_fn(y_preds, y_targets):
+def roc_auc_compute_fn(y_preds: Any, y_targets: Any):
```
Comment (vfdev-5) [inline ignite/contrib/metrics/roc_auc.py:6]: Same here

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @RaviTezu 
I left some comment to improve the type hints

File: ignite/contrib/metrics/average_precision.py (hunk 1 above)
Comment (RaviTezu) [reply ignite/contrib/metrics/average_precision.py:6]: Done

**Review by vfdev-5 — COMMENTED** [review]
Thanks @RaviTezu !
I pushed a minor update myself and if CI is OK we can merge it 🎉

**Review by vfdev-5 — APPROVED** [review]

**Comment by RaviTezu** [comment]
@vfdev-5 Looks like the checks are passed? `(ubuntu-latest, 3.8, pytorch-nightly)` is failing with some runtime socket exception and I am not sure why the `codecov/project` is not happy.

**Comment by vfdev-5** [comment]
@RaviTezu thanks for pinging on this PR. Let me merge it.

> (ubuntu-latest, 3.8, pytorch-nightly) is failing with some runtime socket exception and I am not sure why the codecov/project is not happy.

Yes, it happens to CI to fail from time to time on distributed tests. We can ignore Codecov for now.

**Pull request merged**